app/library/postgres.py

```diff
@@ -102,9 +102,22 @@ async def get_assets_from_postgres(session):
 
 
 async def delete_asset_from_postgres(asset_id: str, session):
-    if not delete_asset(asset_id, session):
+    if not get_asset(asset_id, session):
         raise ValueError("Asset ID not valid")
 
+    scenes_resp = await get_scenes_from_postgres(session)
+    for scene in scenes_resp["scenes"]:
+        if asset_id == str(scene["background_id"]):
+            raise AssertionError(
+                f"Asset ID {asset_id} is reference as background_id in scene with ID {scene['id']}"
+            )
+        for object_id in scene["object_ids"]:
+            object_data = get_object(object_id, session).as_dict()
+            if asset_id == str(object_data["asset_id"]):
+                await delete_object_in_postgres(scene["id"], object_id, session)
```

Comment:
Rather than doing it this way, can you investigate if we can use CASCADE delete in Postgres? CASCADE delete is usually used for this exact reason: when there is a foreign key referencing a key from another table and that other table's key is deleted, then the foreign key references should also be dropped.
https://kb.objectrocket.com/postgresql/how-to-use-the-postgresql-delete-cascade-1369
https://docs.sqlalchemy.org/en/14/orm/cascades.html
If you explore this and find that this is too much work, then ignore this comment. I recognize this might be too much work since we have to remove the object from scene's objects table too. So you may need to explore this first!

Reply:
Looked into CASCADE delete. It is easy to do it for the objects alone but because we also remove from the scene's object array it does get trickier as you mentioned.

I could think of 2 ways with our current schema:
1. We can leave it as is without cascade
2. We use CASCADE delete to get rid of the objects related to the asset first. And then we loop through the scenes and update it's object list to get rid of objects that no longer exist in the Postgres `objects` table.

With option 2 we do save on making multiple calls to the db for deleting objects because of cascade.
Also, if a `scene.object_ids` list is edited then there may be some objects in the database not being used at all. Option 1 would leave them there, option 2 would delete those.

Do you have any thoughts/suggestions?

Reply:
Option 2 sounds good to me, we want to make fewer calls to DB but also make sure that our database stays consistent. You can do a query for get_all_objects after CASCADE delete command, and create a set of the IDs. Then loop through the scenes removing any that have an object_id that isn’t in the set of IDs.

Reply:
Sounds good. I don't think we have a get_all_objects. Do you think it is worth to have a db function to get_all_objects (the whole objects)? Otherwise, I'm thinking to have it just return the object IDs.

Reply:
Yeah we don’t need the entire object, just IDs. I’m sure sqlalcheny has a way to only return certain columns rather than entire table, so maybe we can look at that?